tuning/data/setup_dataprocessor.py

```diff
@@ -0,0 +1,248 @@
+        processor = get_dataprocessor(
+            dataloaderconfig=DataLoaderConfig(), tokenizer=None
+        )
+        data = processor.load_dataset(None, splitName="train[:1]", datafile=data)
+
+    return ("input_ids" in data.column_names) and ("labels" in data.column_names)
+
+
+# For now assume only training dataset is passed via data config file.
+# This is very limited but is done to keep first implementation minimal
+def _process_dataconfig_file(
+    data_args: DataArguments, tokenizer: AutoTokenizer, packing: bool, max_seq_len: int
+):
+    data_config = load_and_validate_data_config(data_args.data_config_path)
+    processor = get_dataprocessor(
+        dataloaderconfig=data_config.dataloader, tokenizer=tokenizer
+    )
+    train_dataset = processor.process_dataset_configs(data_config.datasets)
+
+    data_collator = get_data_collator(
+        packing,
+        data_args.response_template,
+        tokenizer,
+        # Note: Its important to recompute this post handling to
+        #       check if we already tokenized the dataset or not.
+        is_pretokenized_dataset(train_dataset),
+        max_seq_len,
+    )
+
+    dataset_kwargs = {}
+    if is_pretokenized_dataset(train_dataset):
+        dataset_kwargs["skip_prepare_dataset"] = True
+
+    ## HACK: For now just assume we take train_dataset via data config
+    return (
+        train_dataset,
+        None,
+        data_args.dataset_text_field,
+        data_collator,
+        max_seq_len,
+        dataset_kwargs,
+    )
+
+
+def process_dataargs(
+    data_args: DataArguments, tokenizer: AutoTokenizer, train_args: TrainingArguments
+):
+    """
+    Args:
+        data_args: tuning.config.configs.DataArguments
+        tokenizer: AutoTokenizer
+        train_args: TrainingArguments
+            Training arguments passed to the library
+            Used for packing and max_seq_length
+    Returns:
+        Tuple(Dataset, Dataset, str, DataCollator, int, Dict)
+            tuple containing train_dataset, eval_dataset, dataset_text_field,
+                data_collator, max_seq_length and dataset_kwargs
+    """
+
+    max_seq_length = min(train_args.max_seq_length, tokenizer.model_max_length)
+    logging.info("Max sequence length is %s", max_seq_length)
+    if train_args.max_seq_length > tokenizer.model_max_length:
+        logging.warning(
+            "max_seq_length %s exceeds tokenizer.model_max_length \
+            %s, using tokenizer.model_max_length %s",
+            train_args.max_seq_length,
+            tokenizer.model_max_length,
+            tokenizer.model_max_length,
+        )
+
+    if data_args.data_config_path:
+        # Data config is specified so our processing path is diverging
+        return _process_dataconfig_file(
+            data_args, tokenizer, train_args.packing, max_seq_length
+        )
+
+    # Create a data processor with default loader config
+    default_loader_config = DataLoaderConfig()
```

Comment:
I thought we are not calling this `DataLoader` anymore, because it conflates with HF's meaning of `DataLoader`?

Reply:
Renamed. thanks for pointing this out.